orc/src/main/java/org/apache/iceberg/orc/OrcMetrics.java

```diff
@@ -158,8 +158,7 @@ private static Metrics buildOrcMetrics(final long numOfRows, final TypeDescripti
           .orElse(null);
     } else if (columnStats instanceof DateColumnStatistics) {
       min = Optional.ofNullable(((DateColumnStatistics) columnStats).getMinimum())
-          .map(minStats -> DateTimeUtil.daysFromDate(
-              DateTimeUtil.EPOCH.plus(minStats.getTime(), ChronoUnit.MILLIS).toLocalDate()))
+          .map(minStats -> DateTimeUtil.daysFromDate(((Date) minStats).toLocalDate()))
```

Comment:
Javadoc for `Date.toLocalDate` says that the LocalDate returned is "in local time zone". It is implemented like this:

```java
return LocalDate.of(getYear() + 1900, getMonth() + 1, getDate());
```

And, those getter methods use the current zone. So this cannot be correct because the value is dependent on the local zone. I think your intent was to account for the adjustment made by ORC that you referenced by saying "Getting the min/max stats for Date types in ORC are returned with milliseconds adjusted to the local timezone."

I think the main problem is that this uses a path where the value is modified by ORC based on the JVM environment, which breaks a fundamental rule in Iceberg. I don't think that we should use a path where the value is modified.

Reply:
> I think your intent was to account for the adjustment made by ORC that you referenced by saying "Getting the min/max stats for Date types in ORC are returned with milliseconds adjusted to the local timezone."

Yes, ORC modifies the internally stored stats and returns the values dependent on the timezone. These methods adjust for that.

> I think the main problem is that this uses a path where the value is modified by ORC based on the JVM environment, which breaks a fundamental rule in Iceberg. I don't think that we should use a path where the value is modified. It's just too hard to guarantee that although the value was modified, we were able to correctly undo it.

I agree, ORC in this respect violates the rule and modifies the stored value. I'm okay with not using this path, and therefore removing Date metrics collection if that's okay. Does that sound good?

Reply:
I'm okay with either of two options:
1. Don't collect date metrics until this is fixed in ORC
2. Use reflection to pull the `minimum` and `maximum` private int fields out of the stats

Option #2 seems reasonable enough to me, since the stats actually store the int min and max values that we're looking for.